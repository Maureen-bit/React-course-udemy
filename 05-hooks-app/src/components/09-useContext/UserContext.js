import { createContext } from 'react'

/* un context es un order higher component ya que va a contener m√°s componentes adentro */
export const UserContext = createContext(null);
