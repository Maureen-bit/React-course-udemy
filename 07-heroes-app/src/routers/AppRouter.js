import React from 'react';
import { Routes, Route, BrowserRouter } from "react-router-dom";
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/login" element={
          <PublicRoutes>
            <LoginScreen />
          </PublicRoutes>
        } />
       {/* Todas las rutas que no coincidan con login después del slash serán manejadas por DashBoardRoutes */}
        <Route path="/*" element={
          /*PrivateRoutes: es un componente que se crea para darle manejo a las rutas que puede ver el usuario
          solo de manera autenticada */
          <PrivateRoutes>
            <DashboardRoutes />
          </PrivateRoutes>
        } />
      </Routes>
    </BrowserRouter>
  )
}