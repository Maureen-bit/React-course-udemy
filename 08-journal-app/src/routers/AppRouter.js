import React from 'react';
import {
    BrowserRouter,
    Routes,
    Route,
    Navigate
  } from "react-router-dom";
import { JournalScreen } from '../components/journal/JournalScreen';
import { useCheckAuth } from '../hooks/useCheckAuth';
import { CheckingAuth } from '../ui';
import { AuthRouter } from './AuthRouter';

export const AppRouter = () => {
 const { status } = useCheckAuth();

  if(status === 'checking') {
    return <CheckingAuth />
  }

  return (
    <BrowserRouter>
        <Routes>
          {
            status === 'authenticated' ?
            <Route path="/*" element={<JournalScreen />} /> : 
            <Route path="/auth/*" element={<AuthRouter />}  />
          }
          {/*  /auth/* -> el * significa que va a navegar en las rutas descendientes que se encuentran en AuthRouter y 
          que no se va a hacer match con /auth exactamente sino solo con las rutas hijas*/}
          <Route path = "/*" element={<Navigate to="/auth/login" replace /> } />
        </Routes>
  </BrowserRouter>
  )
}
