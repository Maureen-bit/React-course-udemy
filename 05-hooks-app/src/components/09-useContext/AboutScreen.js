import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  const handleClick = () => {
    /* Cuando el context sufre una modificaci√≥n notifica a todos los hijos sobre el cambio y hace que se 
    redibuje cada una de las partes afectadas */
    setUser({})
  }
  return (
    <div>
        <h1>AboutScreen</h1>
        <hr/>
        <pre>
            {JSON.stringify(user, null, 3)}
        </pre>
        <button className='btn btn-warning' 
        onClick={handleClick}
        > 
          Logout 
        </button>
    </div>
  )
}
