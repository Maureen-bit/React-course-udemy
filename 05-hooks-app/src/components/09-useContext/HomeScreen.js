import React, {useContext} from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {
  /* Obtener el valor de nuestro useContext creado para guardar datos del usuario */
  const { user } = useContext(UserContext);
  return (
    <div>
        <h1>HomeScreen</h1>
        <hr/>
        <pre>
            {JSON.stringify(user, null, 3)}
        </pre>
    </div>
  )
}
