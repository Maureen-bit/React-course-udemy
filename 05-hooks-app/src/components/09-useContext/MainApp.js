import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {
    /* UserContext: va a envolver todos los componentes que encuentre en AppRouter, es decir que, la información
    que se guarde en este context estará disponible para todos estos hijos.
    Lo que se encuentra en la propiedad value es lo que voy a compartir  */
    const [user, setUser] = useState({})

  return (
    <UserContext.Provider value={{user, setUser}}>
        <AppRouter />
    </UserContext.Provider>
  )
}
