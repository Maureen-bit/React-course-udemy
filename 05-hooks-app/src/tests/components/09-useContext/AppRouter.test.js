import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Test in <AppRouter /> Components', () => {
    const user = {
        name: 'Maureen',
        email: 'maureen@gmail.com'
    }
    const wrapper = mount( 
    <UserContext.Provider value={{user}}>
       <AppRouter />
    </UserContext.Provider>  );
    test('should first', () => { 
        expect(wrapper).toMatchSnapshot();
     })
 })