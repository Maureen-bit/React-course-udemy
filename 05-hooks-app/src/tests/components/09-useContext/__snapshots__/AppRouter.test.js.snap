// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AppRouter /> Components should first 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        UseContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            exact="true"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            exact="true"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            exact="true"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "name": "Maureen",
   "email": "maureen@gmail.com"
}
      </pre>
    </div>
  </div>
</div>
`;
